consensus/istanbul/core/roundstate_db.go

```diff
@@ -121,35 +139,34 @@ func newRoundStateDB(path string, opts *RoundStateDBOptions) (RoundStateDB, erro
 }
 
 // newMemoryDB creates a new in-memory node database without a persistent backend.
-func newMemoryDB() (*leveldb.DB, error) {
-	db, err := leveldb.Open(storage.NewMemStorage(), nil)
-	if err != nil {
-		return nil, err
-	}
-	return db, nil
+func newMemoryDB() (*leveldb.Database, error) {
+	return leveldb.NewInMemory()
 }
 
 // newPersistentNodeDB creates/opens a leveldb backed persistent node database,
 // also flushing its contents in case of a version mismatch.
-func newPersistentDB(path string) (*leveldb.DB, error) {
-	opts := &opt.Options{OpenFilesCacheCapacity: 5}
-	db, err := leveldb.OpenFile(path, opts)
-	if _, iscorrupted := err.(*lvlerrors.ErrCorrupted); iscorrupted {
-		db, err = leveldb.RecoverFile(path, nil)
-	}
+func newPersistentDB(path string, namespace string) (*leveldb.Database, error) {
+	db, err := leveldb.NewCustom(path, namespace, func(options *opt.Options) {
+		options.BlockSize = 128 * opt.KiB
```

Comment:
If it makes sense, maybe add a small justification for the specific values?

Reply:
Added a comment.